leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/impl/InMemoryMessageStore.java

```diff
@@ -84,19 +98,26 @@ public void deleteFirst(String endpoint) {
      * @param endpoint client's endpoint
      * @return list of queue request entities in order of processing.
      */
-    public List<QueuedRequest> retrieveAll(String endpoint) {
+    public List<QueuedRequest> retrieveAll(final String endpoint) {
         LOG.debug("Retrieve all for endpoint {}", endpoint);
-        BlockingQueue<QueuedRequest> requests = getMessageQueueForEndpoint(endpoint);
-        if (requests.isEmpty()) {
-            return Collections.emptyList();
+        if (queueExist(endpoint)) {
+            return new ArrayList<>(requestQueueMap.get(endpoint));
         } else {
-            return new ArrayList<>(requests);
+            return Collections.emptyList();
         }
     }
 
-    private BlockingQueue<QueuedRequest> getMessageQueueForEndpoint(String endpoint) {
-        return requestQueueMap.get(endpoint) != null ? requestQueueMap.get(endpoint)
-                : new LinkedBlockingQueue<QueuedRequest>();
+    private boolean queueExist(final String endpoint) {
+        return requestQueueMap.get(endpoint) != null ? true : false;
+    }
+
+    /**
+     * For test onyl.
+     * 
+     * @return ConcurrentMap<String, BlockingQueue<QueuedRequest>>
+     */
+    protected ConcurrentMap<String, BlockingQueue<QueuedRequest>> getRequestQueueMap() {
+        return this.requestQueueMap;
```

Comment:
I don't really like the idea to expose the requestQueueMap just for testing.
Maybe we can add a getSize() on this class, with retrieveAll.getSize(), we will cover all your needs.